### ISSUE TITLE
Message 'Unrecognized type <<class 'ansible.template.AnsibleUndefined'>>' when using filter ipaddr

### ISSUE THREAD

**Issue by chemnic** [ansible/ansible-lint#2468]
##### Summary

Latest ansible-lint 6.6.0 shows false positive when using 'ipaddr' filter with a var, but not when using directly a string.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```
ansible-lint 6.6.0 using ansible 2.12.9
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

```yaml
---
- name: Play
  hosts: localhost
  gather_facts: false

  tasks:
    # This PASS
    - name: Debug
      ansible.builtin.debug:
        msg: "{{ '127.0.0.1' | ansible.utils.ipaddr }}"

    # This FAILS
    - name: Debug
      ansible.builtin.debug:
        msg: "{{ somevar | ansible.utils.ipaddr }}"
      vars:
        somevar: '127.0.0.1'
...
```
Result:
```bash
WARNING  Listing 1 violation(s) that are fatal
jinja: Unrecognized type <<class 'ansible.template.AnsibleUndefined'>> for ipaddr filter <value> (jinja[invalid])
playbook_misc/test40.yml:13 Task/Handler: Debug

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - jinja[invalid]  # Rule that looks inside jinja2 templates.

Finished with 1 failure(s), 0 warning(s) on 1 files.
```

**Comment by ssbarnea** [comment]
Fixed in https://github.com/ansible/ansible-lint/releases/tag/v6.6.1

**Comment by chemnic** [comment]
I don't think those changes cover this issue. The problem is the same in 6.6.1:

```console
carlos@it:~$ python -m pip install -U --no-cache-dir ansible-lint
```
```console
carlos@it:~$ ansible-lint --version
ansible-lint 6.6.1 using ansible 2.12.9
```
```console
carlos@it:~$ ansible-lint ./playbook_misc/test40.yml
WARNING  Listing 1 violation(s) that are fatal
jinja: Unrecognized type <<class 'ansible.template.AnsibleUndefined'>> for ipaddr filter <value> (jinja[invalid])
playbook_misc/test40.yml:13 Task/Handler: Debug

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - jinja[invalid]  # Rule that looks inside jinja2 templates.

Finished with 1 failure(s), 0 warning(s) on 1 files.
```

**Comment by ssbarnea** [comment]
At least with Ansible 2.13, I only see a runtime warning from the linter but not error:
```
[WARNING]: an unexpected error occurred during Jinja2 environment setup: unable to locate collection ansible.utils
```

We do only test the linter with latest ansible-core.

**Comment by chemnic** [comment]
I also tested with 2.13.4 with the same results.

To see the error you have to install collection `ansible.utils` and python module `netaddr`. Both are needed to use `ipaddr` filter.

```console
carlos@it:~$ ansible-galaxy collection install ansible.utils
carlos@it:~$ python -m pip install netaddr
```

**Comment by chemnic** [comment]
So, is this a problem with the filter or collection, not with ansible-lint?

I just want to know how to address this before opening an issue somewhere else or using `noqa`.

**Comment by Urth** [comment]
All of the jinja2 filters that are included with ansible raise errors as a `AnsibleFilterError`. But the `ansible.utils` collection raises them as `AnsibleError` which is the generic/base error class defined by ansible. It would probably be best for collections to use `AnsibleFilterError` to raise errors in filters. This allows ansible-lint to catch them specifically.

**Comment by kpfleming** [comment]
I've got version 2.9.0 of ansible.utils installed and I'm still seeing this exact problem.

**Comment by itfors** [comment]
This fails for me:

```
---
- name: Play
  hosts: localhost
  gather_facts: false

  tasks:
    # This FAILS
    - name: Debug
      ansible.builtin.debug:
        msg: "{{ansible_ip_addresses | ansible.utils.ipv4('ipv4') | ansible.utils.ipaddr | first | to_json}}"

```

```
# ansible-playbook -i /dev/null test_ipv4.yml
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [Play] ******************************************************************************************************************

TASK [Debug] *****************************************************************************************************************
fatal: [localhost]: FAILED! => {"msg": "Unrecognized type <<class 'ansible.template.AnsibleUndefined'>> for ipv4 filter <value>"}

PLAY RECAP *******************************************************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0

```

```
# ansible-playbook --version
ansible-playbook [core 2.15.1]
  config file = None
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.9/dist-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible-playbook
  python version = 3.9.2 (default, Feb 28 2021, 17:03:44) [GCC 10.2.1 20210110] (/usr/bin/python3)
  jinja version = 3.1.2
  libyaml = True
```
